twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -0,0 +1,271 @@
+load("//scala:scala.bzl", "scala_mvn_artifact", "scala_library", "write_manifest", "implicit_deps")
+
+def twitter_scrooge():
+  native.maven_jar(
+    name = "libthrift",
+    artifact = "org.apache.thrift:libthrift:0.8.0",
+    sha1 = "2203b4df04943f4d52c53b9608cef60c08786ef2",
+  )
+  native.maven_jar(
+    name = "scrooge_core",
+    artifact = scala_mvn_artifact("com.twitter:scrooge-core:4.6.0"),
+    sha1 = "84b86c2e082aba6e0c780b3c76281703b891a2c8",
+  )
+
+  #scrooge-generator related dependencies
+  native.maven_jar(
+    name = "scrooge_generator",
+    artifact = scala_mvn_artifact("com.twitter:scrooge-generator:4.6.0"),
+    sha1 = "cacf72eedeb5309ca02b2d8325c587198ecaac82",
+  )
+  native.maven_jar(
+    name = "scopt",
+    artifact = scala_mvn_artifact("com.github.scopt:scopt:3.3.0"),
+    sha1 = "9e6fb062b2f12a7945b9f38ee54784aee7edde20",
+  )
+  native.maven_jar(
+    name = "util_core",
+    artifact = scala_mvn_artifact("com.twitter:util-core:6.33.0"),
+    sha1 = "bb49fa66a3ca9b7db8cd764d0b26ce498bbccc83",
+  )
+  native.maven_jar(
+    name = "util_logging",
+    artifact = scala_mvn_artifact("com.twitter:util-logging:6.33.0"),
+    sha1 = "3d28e46f8ee3b7ad1b98a51b98089fc01c9755dd",
+  )
+
+def _collect_transitive_srcs(targets):
+  r = set()
+  for target in targets:
+    if hasattr(target, "thrift"):
+      r += target.thrift.transitive_srcs
+  return r
+
+def _collect_owned_srcs(targets):
+  r = set()
+  for _target in targets:
+    if hasattr(_target, "extra_information"):
+      for target in _target.extra_information:
+        if hasattr(target, "scrooge_srcjar"):
+          r += target.scrooge_srcjar.transitive_owned_srcs
+  return r
+
+#TODO this is the same as in scala.bzl...just depend on that one?
```

Comment:
actually, since I wrote that comment it's not the same anymore... I'll split it into the shared and not